src/Core/Models/Business/SubscriptionUpdate.cs

```diff
@@ -63,22 +65,24 @@ public override List<SubscriptionItemOptions> UpgradeItemsOptions(Subscription s
 
         public override List<SubscriptionItemOptions> RevertItemsOptions(Subscription subscription)
         {
+
             var item = SubscriptionItem(subscription, PlanIds.Single());
             return new()
             {
                 new SubscriptionItemOptions
                 {
                     Id = item?.Id,
                     Plan = PlanIds.Single(),
-                    Quantity = _organization.Seats,
-                    Deleted = item?.Id != null ? true : (bool?)null,
+                    Quantity = _previousSeats,
+                    Deleted = _previousSeats == 0 ? true : (bool?)null,
                 }
             };
         }
     }
 
     public class StorageSubscriptionUpdate : SubscriptionUpdate
     {
+        private Nullable<long> _prevStorage;
```

Comment:
I propose just using `long?` here instead of the full name.

Reply:
Done